src/ref_serializer.rs

```diff
@@ -139,107 +149,114 @@ impl<'a> RefListSerializer<'a> {
 ///         "member3",
 ///         Decimal::from_f64(12.34566).unwrap(),
 ///    )
+///    .unwrap()
+///    .finish()
 ///    .unwrap();
 /// assert_eq!(
-///    serialized_item,
+///    serialized_dict,
 ///    "member1=11;foo, member2=(abc;abc_param=?0 def);bar=\"val\", member3=12.346"
 /// );
 /// ```
 #[derive(Debug)]
-pub struct RefDictSerializer<'a> {
-    buffer: &'a mut String,
+pub struct RefDictSerializer {
+    buffer: String,
 }
 
-impl<'a> RefDictSerializer<'a> {
-    pub fn new(buffer: &'a mut String) -> Self {
-        RefDictSerializer { buffer }
+impl RefDictSerializer {
+    pub fn new() -> Self {
+        RefDictSerializer {
+            buffer: String::new(),
+        }
     }
 
-    pub fn bare_item_member(self, name: &str, value: impl AsRefBareItem) -> SFVResult<Self> {
-        if !self.buffer.is_empty() {
-            self.buffer.push_str(", ");
-        }
-        Serializer::serialize_key(name, self.buffer)?;
+    pub fn bare_item_member(mut self, name: &str, value: impl AsRefBareItem) -> SFVResult<Self> {
+        write_separator(&mut self.buffer);
+        Serializer::serialize_key(name, &mut self.buffer)?;
         let value = value.as_ref_bare_item();
         if value != RefBareItem::Boolean(true) {
             self.buffer.push('=');
-            Serializer::serialize_bare_item(value, self.buffer)?;
+            Serializer::serialize_bare_item(value, &mut self.buffer)?;
         }
         Ok(self)
     }
 
-    pub fn parameter(self, name: &str, value: impl AsRefBareItem) -> SFVResult<Self> {
+    pub fn parameter(mut self, name: &str, value: impl AsRefBareItem) -> SFVResult<Self> {
         if self.buffer.is_empty() {
             return Err("parameters must be serialized after bare item or inner list");
         }
-        Serializer::serialize_parameter(name, value, self.buffer)?;
-        Ok(RefDictSerializer {
-            buffer: self.buffer,
-        })
+        Serializer::serialize_parameter(name, value, &mut self.buffer)?;
+        Ok(self)
     }
 
-    pub fn open_inner_list(self, name: &str) -> SFVResult<RefInnerListSerializer<'a, Self>> {
-        if !self.buffer.is_empty() {
-            self.buffer.push_str(", ");
-        }
-        Serializer::serialize_key(name, self.buffer)?;
+    pub fn open_inner_list(mut self, name: &str) -> SFVResult<RefInnerListSerializer<Self>> {
+        write_separator(&mut self.buffer);
+        Serializer::serialize_key(name, &mut self.buffer)?;
         self.buffer.push_str("=(");
-        Ok(RefInnerListSerializer::<RefDictSerializer> {
-            buffer: self.buffer,
-            caller_type: PhantomData,
-        })
+        Ok(RefInnerListSerializer { buffer: self })
+    }
+
+    pub fn finish(self) -> SFVResult<String> {
+        if self.buffer.is_empty() {
+            return Err("serializing empty dictionary is not allowed");
```

Comment:
Similar to the other PR, I don't think this is something that we need to enforce?

Reply:
See https://github.com/undef1nd/sfv/issues/132#issuecomment-2671490077.